### ISSUE TITLE
Unexpected nullable warning on Property initialisation and [AllowNull]

### ISSUE THREAD

**Issue by LokiMidgard** [dotnet/roslyn#39966]
**Version Used**: 
VS 16.3.10
**Steps to Reproduce**:

```c#
using System;
using System.Diagnostics.CodeAnalysis;

namespace ConsoleApp
{

    public class Foo
    {
        private string? bar;

        [AllowNull]
        public string Bar { get => this.bar ?? string.Empty; set => this.bar = value; }

        static void Main(string[] args)
        {
            var f1 = new Foo()
            {
                Bar = null // CS8625 
            };
            var f2 = new Foo();
            f2.Bar = null; // No Warning
        }

    }
}
```
**Expected Behavior**:
f1.Bar will be assigned without warning in the propertery initialisation.
**Actual Behavior**:
Shows a Warning in property initilizer.

**Comment by RikkiGibson** [comment]
Thanks for reporting this bug @LokiMidgard!
